.changeset/rare-lemons-study.md

```diff
@@ -0,0 +1,31 @@
+---
+'@vanilla-extract/recipes': patch
+---
+
+Add `RecipeVariants` type to get variants from a recipe.
+
+This type would help consumers easily get the variants type. Useful for typing their components:
+
+```tsx
+const textRecipes = recipe({
+  variants: {
+    size: {
+      small: { fontSize: '12px' },
+      medium: { fontSize: '16px' },
+      large: { fontSize: '24px' },
+    },
+  },
+});
+
+type TextVariants = RecipeVariants<typeof textRecipes>;
+
+type Props = {
+  // ... other props
+} & TextVariants;
+
+function Text({ size, ...props }: Props) {
+  return <div {...props} className={textRecipes({ size })} />;
+}
+
+<Text size="small">Awesome text</Text>;
```

Comment:
We try to avoid framework specific examples in docs and changesets. Would be great to just use the example that you wrote in the docs here.